### PR TITLE
flake8: fix PEP8 warnings about lambda

### PR DESCRIPTION
**Pull request by ZenithDK**
Fix the 'lambda to def' warnings

### CODE REVIEW

**Comment by jodal** [comment]
You need to review the functions and add `return` to the ones where the lambda return value was used for anything. That's probably the reason for the test failures.

File: tests/mpd/test_commands.py [hunk 1]
```diff
@@ -111,12 +115,16 @@ def test_function_has_no_arguments_fails(self):
 
     def test_function_has_required_and_varargs_fails(self):
         with self.assertRaises(TypeError):
-            func = lambda context, required, *args: True
+            def func(context, required, *args):
+                True
```
Comment (jodal) [inline tests/mpd/test_commands.py]: I guess this and the next function could simply have `pass` as the body. With `True` I was wondering a bit if the function body had any relevance to the test.

File: tests/mpd/test_commands.py [hunk 2]
```diff
@@ -203,28 +222,38 @@ def test_validator_skips_optional_default(self):
...
+                return wrong_arg
+
             self.commands.add('bar', arg=lambda v: v)(func)
 
     def test_validator_called_context_fails(self):
         return  # TODO: how to handle this
         with self.assertRaises(TypeError):
-            func = lambda context: True
+            def func(context):
+                True
```
Comment (jodal) [inline tests/mpd/test_commands.py]: Same here.

**Comment by jodal** [comment]
Didn't notice that you had force-pushed a fixed commit until I went checking the PR now. Please comment when you update the PR, as GitHub doesn't send mail notifications on new code in a PR. :-)

**Comment by ZenithDK** [comment]
My mistake, would you prefer I add more commits iterating on a fix instead of rebasing and force-pushing?

**Comment by jodal** [comment]
Nah, I'm a sucker for a clean git history :-)

The GitHub PR notification behavior is the same either way.

**Comment by ZenithDK** [comment]
I have changed to "pass" in those two places and in two other places.

**Comment by ZenithDK** [comment]
For real now ;-)

File: tests/mpd/test_commands.py [hunk 3]
```diff
@@ -64,7 +64,8 @@ def test(context):
             pass
 
     def test_register_second_command_to_same_name_fails(self):
-        func = lambda context: True
+        def func(context):
+            True
```
Comment (jodal) [inline tests/mpd/test_commands.py]: One :-)

File: tests/mpd/test_commands.py [hunk 4]
```diff
@@ -203,28 +222,38 @@ def test_validator_skips_optional_default(self):
...
         self.commands.add('bar', arg=validdate)(func)
 
         with self.assertRaises(exceptions.MpdArgError):
             self.commands.call(['bar', 'test'])
 
     def test_auth_required_gets_stored(self):
-        func1 = lambda context: context
-        func2 = lambda context: context
+        def func1(context):
+            context
```
Comment (jodal) [inline tests/mpd/test_commands.py]: Two :-)

File: tests/mpd/test_commands.py [hunk 5]
```diff
@@ -203,28 +222,38 @@ def test_validator_skips_optional_default(self):
...
+
+        def func2(context):
+            context
```
Comment (jodal) [inline tests/mpd/test_commands.py]: Three :-)

File: tests/mpd/test_commands.py [hunk 6]
```diff
@@ -232,8 +261,12 @@ def test_auth_required_gets_stored(self):
         self.assertFalse(self.commands.handlers['bar'].auth_required)
 
     def test_list_command_gets_stored(self):
-        func1 = lambda context: context
-        func2 = lambda context: context
+        def func1(context):
+            context
```
Comment (jodal) [inline tests/mpd/test_commands.py]: Four :-)

File: tests/mpd/test_commands.py [hunk 7]
```diff
@@ -232,8 +261,12 @@ def test_auth_required_gets_stored(self):
...
+
+        def func2(context):
+            context
```
Comment (jodal) [inline tests/mpd/test_commands.py]: Five :-)

**Comment by jodal** [comment]
There's still five comments to address here! :-)

**Comment by ZenithDK** [comment]
Duh, I missed a "git add -u" before force pushing.
Thanks for noticing!

File: mopidy/local/search.py [hunk 8]
```diff
@@ -102,38 +117,56 @@ def search(tracks, query=None, uris=None):
...
-                artist_filter(t) or
-                albumartist_filter(t) or
-                composer_filter(t) or
-                performer_filter(t) or
-                track_no_filter(t) or
-                genre_filter(t) or
-                date_filter(t) or
-                comment_filter(t))
+            def uri_filter(t):
+                    return bool(t.uri and q in t.uri.lower())
```
Comment (jodal) [inline mopidy/local/search.py]: There's an extra level of indentation here.

**Comment by jodal** [comment]
+1, with 1 comment

**Comment by ZenithDK** [comment]
How the heck did flake8 not notice that extra level of indentation? Fixed now.

**Pull request merged**